File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        nums_and_indexes = {}
        for i in range(len(nums)):
            if target - nums[i] in nums_and_indexes:
                return [i, nums_and_indexes[target - nums[i]]]
            else:
                nums_and_indexes[nums[i]] = i
        
        return []

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> numAndIndex;
        for(int i = 0; i < nums.size(); i++)
        {
            auto curr = numAndIndex.find(target - nums[i]);
            if(curr != numAndIndex.end())
                return { curr->second, i };
            numAndIndex[nums[i]] = i;
        }
        return {};
    }
};